pkg/oc/admin/diagnostics/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -42,7 +41,10 @@ type AggregatedLogging struct {
 	DCClient          appstypedclient.DeploymentConfigsGetter
 	SCCClient         securitytypedclient.SecurityContextConstraintsGetter
 	KubeClient        kclientset.Interface
-	result            types.DiagnosticResult
+	sumResult         types.DiagnosticResult
+	projResult        map[string]types.DiagnosticResult
+	currentProject    string
+	CmdlineProject    string
```

Comment:
Do not consider user provided project as special case.
I think this will be clean:

- Introduce optional flag
```
flagLoggingProjects = "logging-projects"  (plural instead of 'logging-project' as we want to use 'logging' and 'openshift-logging' as default projects)
```

- Show default projects in the aggregatedlogging subcommand:
```
defaultLoggingProjects := []string{"logging", "openshift-logging"}
[]types.Parameter{
   {flagLoggingProjects, "Comma separated list of projects that support logging", &d.LoggingProjects, strings.Join(defaultLoggingProjects, ",")},
}
```

- Set d.LoggingProjects variable if required before performing checks
```
if len(d.LoggingProjects) == 0 {
   d.LoggingProjects = defaultLoggingProjects
}
```
- Perform checks for each logging project
```
for _, project := range d.LoggingProjects {
    // Skip the project if it does't contain logging
    // Perform diagnostic: d.checkProjectDiagnostics
    // Log errors or warnings
}
```

Reply:
```
flagLoggingProjects = "logging-projects"  (plural instead of 'logging-project' as we want to use 'logging' and 'openshift-logging' as default projects)
```

that contradicts what you said earlier:

> Add '--logging-project' optional param to aggregatedlogging sub-command.
> If the optional param is not provided then check for 'logging' or 'openshift-logging' project.

Reply:
Okay, If we don't allow more than one logging deployment per cluster then '--logging-project' looks good. 
Contemplating more on this, we do not have to loop over both "logging" and "openshift-logging" projects.
- Add optional '--logging-project', if not specified default to 'logging' (when we switch to 'openshift-logging', we can also update default value for this param).
- As mentioned before, do not consider user provided logging project as special case. 
```
 type AggregatedLogging struct {
  ...
    LoggingProject  string
}

defaultLoggingProject := "logging"

{flagLoggingProject, "Project that supports aggregated logging", &d.LoggingProject, defaultLoggingProject} 

// d.LoggingProject will have user defined project if set otherwise will have default value (currently 'logging', later it could be 'openshift-logging')
```
This makes the implementation clean and simple.
- Lets say, admin/user deploys 'logging' project and later updated new version of openshift which has 'openshift-logging' as default logging. In this case they can always pass --logging-project='logging'.

Reply:
I disagree again, we discussed it with Jeff and we want the default cases to work out-of-the-box. Both 'logging' and 'openshift-logging' **MUST** work without the user specifying **ANYTHING**, it is the extra convenience for unusual setups that `--logging-project` flag provides but nothing else.

Reply:
If we want this to work in both cases, then we could say default value as "logging or openshift-logging" (seems bit ugly but I don't see any other alternative).
Still don't see the need for looping over the projects. We know ONLY one logging project is allowed in the cluster. We could check if 'openshift-logging' (gets precedence) or 'logging' project exists and set d.LoggingProject to that project if not specified by user.

Reply:
I am afraid I don't understand what are you trying to say. This doesn't loop over the projects. The implementation is rather simple:

- check `logging` project
  - if no errors found, great, we have diagnosed perfectly configured EFK stack, return no errors
- check `openshift-logging` project
  - if no errors found, great, we have diagnosed perfectly configured EFK stack, return no errors
- we don't have any more defaults to look into, report errors for both because we don't have a clue which is the EFK stack and which is user-defined project at the moment
- allow `--logging-project` flag to take custom defined project because currently, we allow EFK to be deployed anywhere 
  - https://github.com/openshift/openshift-ansible/blob/d5879135f077e4aaaa09c8e4ebf5d9ef2a063a78/roles/openshift_logging/tasks/install_logging.yaml#L13